### DISCUSSION TITLE
ReSpeaker USB Mic Array | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1129] [Q&A]
# ReSpeaker USB Mic Array | Seeed Studio Wiki

ReSpeaker USB Mic Array

https://wiki.seeedstudio.com/ReSpeaker-USB-Mic-Array/

**Comment by 213124ds** [comment]
After odas is started, the channel will be closed automatically, and how to change the network settings from IPv6 to IPv4? My current situation is that 9000.9001 is IPv4, 10000, and 10010 are IPv6, but there are no settings for 10000 and 10010 in the server.js file.

**Comment by jconnell11** [comment]
Super simple [ROS node](https://github.com/jconnell11/Wansui/blob/main/wansui_vis/scripts/dir_mic.py) in Python that basically just copies lights. Topic "sound_dir" is 0 if no sound, else 1-360 for degrees CCW for source direction with USB plug at 180 degrees. Import "mic_tuning" is just renamed "tuning.py" from "usb_4_mic_array" download. Note that you have to add a USB access permission file (as described in source) then reboot.

**Reply by nfs0619** [reply]
Thank you for your valuable input! We highly encourage developers like you to contribute to our project. If you’re interested in getting more involved, here is more information on how to become a contributor:

[Contributor Project Page](https://github.com/orgs/Seeed-Studio/projects/6/views/1?pane=issue&itemId=30957479)
[Contributors Introduction](https://wiki.seeedstudio.com/contributors/)
Looking forward to your participation!

**Comment by Aanistratenko1984** [comment]
Hi,
We purchased the ReSpeaker USB Mic Array for voice recording and subsequent recognition.
We need the declared feature "Speech algorithms and features - De-reverberation."
https://wiki.seeedstudio.com/ReSpeaker-USB-Mic-Array/#features
However, the list of parameters in tuning.py does not include any parameters related to reverberation.
How can we enable and configure the de-reverberation feature?

**Reply by nfs0619** [reply]
Hello,

The de-reverberation feature is built into the firmware and is automatically enabled. It cannot be modified through the parameters in `tuning.py`. This feature helps reduce echo and reverberation during recording.

If you need further assistance, feel free to contact our technical support team at [techsupport@seeed.io](mailto:techsupport@seeed.io).

**Comment by muhammadxsalman** [comment]
Can we use this with nvidia jetson orin nx? Or if there is any other microphone available with same features (built-in AEC, VAD, DOA, Beamforming and NS)?

**Reply by HM-sur** [reply]
sorry, but it doesn't fit right now

**Comment by anamtya** [comment]
Hi,  I am using a 4mic usb array,  
I notice that there seems to be a default low pass filter on with a cutoff at 7K hz, how can I make it record 20-20k hz?
Also can i get into the firmware and increase the sample rate?

**Reply by nfs0619** [reply]
Thanks for your question! For detailed technical support on this, we recommend reaching out to our team at [techsupport@seeed.io](mailto:techsupport@seeed.io).

**Comment by HomeyRichy** [comment]
For this Respeaker USB unit I see my Raspberry Pi 5 under heavy load. 

1.	Would an update to the 1 channel firmware be a solution giving less traffic?
2.	How can I update firmware? The  info on this page uses pyusb library.
This lib is giving problems with python on usb.core.

The script dfu.py isn't working with error on import usb.core

Help is welcome!

**Reply by HM-sur** [reply]
hi,
Do you encounter the following error when flushing firmware on Pi5
```
(myenv) seeed@raspberrypi:~/usb_4_mic_array $ sudo python3 dfu.py --download 1_channel_firmware.bin
Traceback (most recent call last):
  File "/home/seeed/usb_4_mic_array/dfu.py", line 15, in <module>
    import usb.core
ModuleNotFoundError: No module named 'usb'
````
This is due to usb permissions, you just need to add permissions. The process is as follows.
Check your device number
```
(myenv) seeed@raspberrypi:~/usb_4_mic_array $ lsusb
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 003: ID 2886:0018 Seeed Technology Co., Ltd. ReSpeaker 4 Mic Array (UAC1.0)
```
The code below should be changed to your device number
```
sudo chmod 666 /dev/bus/usb/003/003
```
Then you can start running flushing
```
python3 dfu.py --download 1_channel_firmware.bin
```
Feel free to ask any questions during the process

**Comment by HomeyRichy** [comment]
Thank you very much for your feedback. I will give this a try, this was one of the problems. Rest has to do with Python versions. I’m using Python3 while these scripts are on version 2 if I’m right.

Maybe just look for a newer hardware is the other option, though I haven’t found a newer alternative. Advantage of USB with soundcard on board, probably lower load on Raspberry and the casing included made me choose this version of reSpeaker.
Suggestions welcome for an Alexa alternative TTS hardware!

**Comment by LeeHieu134** [comment]
Hi All,
After flashing with:

sudo python dfu.py --download MicArrayV3_firmware/6_channels_dfu_4.0.0_firmware.bin

my device no longer enumerates in DFU mode. How can I recover/force it back into DFU? I tried:
sudo python dfu.py --download 6_channels_firmware.bin
and it didn't work. I'm using v2.0 and just want it to work like it did initially.
please help me!

**Reply by nfs0619** [reply]
Hi,
We recommend that you contact our technical support team at **[techsupport@seeed.io](mailto:techsupport@seeed.io)**. They will be able to provide you with faster and more detailed guidance.
